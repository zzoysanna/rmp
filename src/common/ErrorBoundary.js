import React, {Component} from 'react';
import './styles/error-boundary.less';

export default class ErrorBoundary extends Component {
	constructor(props) {
		super(props);
		this.state = {hasError: false};
	}

	static getDerivedStateFromError() {
		return { hasError: true };
	}

	render() {
		if (this.state.hasError) {
			return (
				<div className="error">
					<img src="https://miro.medium.com/max/800/1*cLOskwDOJD2FWB9xbYF7aQ.gif" alt="have a break"/>
					<h1>Sorry, something went wrong</h1>
				</div>
			);
		}

		return this.props.children;
	}
}
