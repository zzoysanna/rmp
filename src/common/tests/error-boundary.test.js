import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from '../ErrorBoundary';

describe('ErrorBoundary', () => {

	beforeAll(() => {
	});

	it('should be render if not errors', () => {
		const component = shallow(<ErrorBoundary />);
		expect(component.state.hasError).toBeFalsy();
		expect(component).toMatchSnapshot();
	});

	it('should be render if has errors', () => {
		const component = shallow(<ErrorBoundary />);
		const instance = component.instance();
		instance.setState({ hasError: true });
		expect(instance.state['hasError']).toBeTruthy();
		expect(component).toMatchSnapshot();
	});


});
