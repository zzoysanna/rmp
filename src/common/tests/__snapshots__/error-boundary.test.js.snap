// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should be render if has errors 1`] = `ShallowWrapper {}`;

exports[`ErrorBoundary should be render if not errors 1`] = `ShallowWrapper {}`;
